dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -560,38 +562,52 @@ public BlockReader createFileReader(String fileId, long offset, boolean position
   }
 
   @Override
-  public ListenableFuture<List<FileFailure>> load(List<File> files, UfsReadOptions options) {
+  public ListenableFuture<List<LoadFileFailure>> load(boolean loadData, List<UfsStatus> ufsStatuses, UfsReadOptions options) {
+    System.out.println("Request: " + Arrays.toString(ufsStatuses.toArray()));
     List<ListenableFuture<Void>> futures = new ArrayList<>();
-    List<FileFailure> errors = Collections.synchronizedList(new ArrayList<>());
-    for (File file : files) {
-      ListenableFuture<Void> loadFuture = Futures.submit(() -> {
-        try {
-          if (options.hasUser()) {
-            AuthenticatedClientUser.set(options.getUser());
+    List<LoadFileFailure> errors = Collections.synchronizedList(new ArrayList<>());
+    for (UfsStatus status : ufsStatuses) {
+      String ufsFullPath = status.getUfsFullPath().toString();
+      DoraMeta.FileStatus fs = buildFileStatusFromUfsStatus(status, ufsFullPath);
+      mUfsStatusCache.put(ufsFullPath, fs);
+      if (mMetaStore != null) {
+        mMetaStore.putDoraMeta(ufsFullPath, fs);
+      }
+      if (loadData && status.isFile() && (status.asUfsFileStatus().getContentLength() > 0)) {
+        ListenableFuture<Void> loadFuture = Futures.submit(() -> {
+          try {
+            if (options.hasUser()) {
+              AuthenticatedClientUser.set(options.getUser());
+            }
+            load(status.getUfsFullPath().toString(),
+                0, status.asUfsFileStatus().getContentLength(), false);
+          } catch (Exception e) {
+            AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
+            errors.add(LoadFileFailure.newBuilder().setUfsPath(status.getUfsFullPath().toString())
+                .setCode(t.getStatus().getCode().value())
+                .setRetryable(t.isRetryable())
+                .setMessage(t.getMessage()).build());
           }
-          load(file.getUfsPath(), file.getMountId(), file.getLength());
-        } catch (Exception e) {
-          AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
-          errors.add(FileFailure.newBuilder().setFile(file).setCode(t.getStatus().getCode().value())
-              .setRetryable(t.isRetryable()).setMessage(t.getMessage()).build());
-        }
-      }, GrpcExecutors.BLOCK_READER_EXECUTOR);
-      futures.add(loadFuture);
+        }, GrpcExecutors.BLOCK_READER_EXECUTOR);
+        futures.add(loadFuture);
+      }
     }
     return Futures.whenAllComplete(futures).call(() -> errors, GrpcExecutors.BLOCK_READER_EXECUTOR);
   }
 
-  private void load(String ufsPath, long mountId, long length) {
+  private void load(String ufsPath, long mountId, long length, boolean reloadMetadata) {
```

Comment:
where r u planning to call this func with reloadMetadata = true?

Reply:
actually instead of avoid load metadata here shouldnt we let the metadata load from ufs happen from here and then do data load?